Выполнил практическое задание спринта №16

MaximusDenisov-YP opened the pull request.

MaximusDenisov-YP committed:
Выполнил практическое задание спринта №16

MaximusDenisov-YP committed:
Исправил замечания Check and Build application

avfyodorov commented on gateway/src/main/java/ru/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,59 @@
+
+    @GetMapping("/search")
+    public ResponseEntity<Object> searchByText(@RequestParam String text) {
+        return itemClient.searchByText(text);
```
Лучше было бы прямо здесь проверить входной параметр text. И, если текст запроса пустой, или недопустимый, например, состоит из пробелов, то прямо отсюда и вернуть результат, не обращаться к серверу. Вернуть можно, например, пустую коллекцию.

avfyodorov commented on gateway/src/main/java/ru/shareit/ShareItGateway.java:
```diff
@@ -0,0 +1,12 @@
+    public static void main(String[] args) {
+        SpringApplication.run(ShareItGateway.class, args);
+    }
+}
```
Пишу здесь, потому что обработчик ошибок сервера недоступен для комментария. 
Вот тут есть один момент.
 
Теперь есть (должно быть) два обработчика ошибок: один - в модуле гейтвей,  а второй - это сервер. И ошибки перехватываются  и обрабатываются, соответственно, разные. Например, на сервере валидация по заданию не нужна( точнее почти не нужна), только на gateway, поэтому и bad request на сервере  вроде бы как быть не должно. И наоборот, на  gateway не должно быть ничего,  имеющего отношение к серверу (not found). 

На сервере только та валидация входных данных , которая требует работы с БД.
А  в остальном на сервере(во всех классах, где ранее была валидация) не должно быть валидации и ее не должно быть и в зависимостях в pom файле тоже.
Есть предложение ещё раз проверить и уточнить, где какие ошибки ловятся и обрабатываются.

avfyodorov reviewed (commented):
Добрый день, Максим!

Практически всё сделано, осталось чуть доделать.

MaximusDenisov-YP committed:
Исправил замечания от ревьюера.

- Добавлен GatewayExceptionHandler.java
- Добавлен ErrorResponse.java для ShareItGateway
- Добавлена проверка в ItemController на параметр метода searchByText()
- Удалены лишние ExceptionHandlers в GlobalExceptionHandler.java в ShareItServer

avfyodorov reviewed (commented):
Добрый вечер, Максим!

Замечаний нет.
Работа принята.
